packages/headless/src/controllers/commerce/pagination/core/headless-core-commerce-pagination.test.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  selectPage,
+  nextPage,
+  previousPage,
+} from '../../../../features/commerce/pagination/pagination-actions';
+import {paginationReducer as commercePagination} from '../../../../features/commerce/pagination/pagination-slice';
+import {fetchProductListing} from '../../../../features/commerce/product-listing/product-listing-actions';
+import {buildMockCommerceEngine, MockCommerceEngine} from '../../../../test';
+import {
+  buildCorePagination,
+  Pagination,
+} from './headless-core-commerce-pagination';
+
+describe('core pagination', () => {
+  let engine: MockCommerceEngine;
+  let pagination: Pagination;
+  const fetchResultsActionCreator = fetchProductListing;
+
+  function initPagination() {
+    engine = buildMockCommerceEngine();
+
+    pagination = buildCorePagination(engine, {fetchResultsActionCreator});
+  }
+
+  beforeEach(() => {
+    initPagination();
+  });
+
+  it('adds correct reducers to engine', () => {
+    expect(engine.addReducers).toBeCalledWith({
+      commercePagination,
+    });
+  });
+
+  it('exposes #subscribe method', () => {
+    expect(pagination.subscribe).toBeTruthy();
+  });
+
+  it('#selectPage dispatches #selectPage & #fetchResultsActionCreator', () => {
```

Comment:
I know I did it this way originally, but now that I look back at it, I think in general it would be a cleaner if we grouped tests by method (unless there's only one thing to test for a method, in which case a single 'it' would suffice) and do one test per expected dispatched action. Also, we should test that the correct payload is being passed with the dispatched action when applicable.

E.g., here we could do:

```typescript
describe('#selectPage', () => {
  let page: number;

  beforeEach(() => {
    page = 0;
    pagination.selectPage(pageToSelect);
  });

  it('dispatches #selectPage with correct payload', () => {
    expect(engine.actions).toContainEqual(selectPage(page));
  });

  it('dispatches #fetchResultsActionCreator' () => {
    const action = engine.findAsyncAction(fetchResultsActionCreator.pending);
    expect(action).toBeTruthy();
  });
});
```